### PR TITLE
Gateway localnet examples

### PR DESCRIPTION
**Pull request by fadeev**
## Summary by CodeRabbit

- **New Features**
	- Introduced `Hello` contract for enhanced cross-chain interaction and structured revert handling.
	- Introduced `RevertContract` for improved revert scenario management.
	- Added `ReceiverContract` for event logging and Ether reception.
	- Added new Hardhat tasks, `call-from-zetachain` and `deposit-and-call`, facilitating seamless interactions with ZetaChain contracts.

- **Documentation**
	- Enhanced documentation to guide users in utilizing new features and interacting with the ZetaChain contracts.

- **Chores**
	- Improved project structure with new management files, contributing to better organization and maintenance.

### CODE REVIEW

**Comment by fadeev** [comment]
```
cd universal/hello
yarn
npx hardhat localnet
```

```
npx hardhat deploy --network localhost
```

```
npx hardhat interact --contract 0x0DCd1Bf9A1b36cE34237eEaFef220932846BCD82 --name alice  --network localhost
```

```
Error calling contract: Error: cannot estimate gas; transaction may fail or may require manual gas limit [ See: https://links.ethers.org/v5-errors-UNPREDICTABLE_GAS_LIMIT ] (reason="execution reverted", method="estimateGas", transaction={"from":"0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266","to":"0x9fE46736679d2D9a65F0992F2272dE9f3c7fa6e0","data":"0x45ac7a8d0000000000000000000000000dcd1bf9a1b36ce34237eeafef220932846bcd82000000000000000000000000000000000000000000000000000000000000006000000000000000000000000000000000000000000000000000000000000000e0000000000000000000000000000000000000000000000000000000000000006000000000000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000000000000000005616c6963650000000000000000000000000000000000000000000000000000000000000000000000000000004955a3f38ff86ae92a914445099caa8ea2b9ba3200000000000000000000000000000000000000000000000000000000000000000000000000000000000000004955a3f38ff86ae92a914445099caa8ea2b9ba3200000000000000000000000000000000000000000000000000000000000000800000000000000000000000000000000000000000000000000000000000000000","accessList":null}, error={"name":"ProviderError","_stack":"ProviderError: execution reverted\n    at HttpProvider.request (/Users/fadeev/github.com/zeta-chain/example-contracts/universal/hello/node_modules/hardhat/src/internal/core/providers/http.ts:90:21)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async EthersProviderWrapper.send (/Users/fadeev/github.com/zeta-chain/example-contracts/universal/hello/node_modules/@nomiclabs/hardhat-ethers/src/internal/ethers-provider-wrapper.ts:13:20)","code":3,"_isProviderError":true,"data":"0x"}, code=UNPREDICTABLE_GAS_LIMIT, version=providers/5.7.2)
    at Logger.makeError (/Users/fadeev/github.com/zeta-chain/example-contracts/universal/hello/node_modules/@ethersproject/logger/src.ts/index.ts:269:28)
    at Logger.throwError (/Users/fadeev/github.com/zeta-chain/example-contracts/universal/hello/node_modules/@ethersproject/logger/src.ts/index.ts:281:20)
    at checkError (/Users/fadeev/github.com/zeta-chain/example-contracts/universal/hello/node_modules/@ethersproject/providers/src.ts/json-rpc-provider.ts:78:20)
    at EthersProviderWrapper.<anonymous> (/Users/fadeev/github.com/zeta-chain/example-contracts/universal/hello/node_modules/@ethersproject/providers/src.ts/json-rpc-provider.ts:642:20)
    at step (/Users/fadeev/github.com/zeta-chain/example-contracts/universal/hello/node_modules/@ethersproject/providers/lib/json-rpc-provider.js:48:23)
    at Object.throw (/Users/fadeev/github.com/zeta-chain/example-contracts/universal/hello/node_modules/@ethersproject/providers/lib/json-rpc-provider.js:29:53)
    at rejected (/Users/fadeev/github.com/zeta-chain/example-contracts/universal/hello/node_modules/@ethersproject/providers/lib/json-rpc-provider.js:21:65)
    at processTicksAndRejections (node:internal/process/task_queues:95:5) {
  reason: 'execution reverted',
  code: 'UNPREDICTABLE_GAS_LIMIT',
  method: 'estimateGas',
  transaction: {
    from: '0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266',
    to: '0x9fE46736679d2D9a65F0992F2272dE9f3c7fa6e0',
    data: '0x45ac7a8d0000000000000000000000000dcd1bf9a1b36ce34237eeafef220932846bcd82000000000000000000000000000000000000000000000000000000000000006000000000000000000000000000000000000000000000000000000000000000e0000000000000000000000000000000000000000000000000000000000000006000000000000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000000000000000005616c6963650000000000000000000000000000000000000000000000000000000000000000000000000000004955a3f38ff86ae92a914445099caa8ea2b9ba3200000000000000000000000000000000000000000000000000000000000000000000000000000000000000004955a3f38ff86ae92a914445099caa8ea2b9ba3200000000000000000000000000000000000000000000000000000000000000800000000000000000000000000000000000000000000000000000000000000000',
    accessList: null
  },
  error: ProviderError: execution reverted
      at HttpProvider.request (/Users/fadeev/github.com/zeta-chain/example-contracts/universal/hello/node_modules/hardhat/src/internal/core/providers/http.ts:90:21)
      at processTicksAndRejections (node:internal/process/task_queues:95:5)
      at async EthersProviderWrapper.send (/Users/fadeev/github.com/zeta-chain/example-contracts/universal/hello/node_modules/@nomiclabs/hardhat-ethers/src/internal/ethers-provider-wrapper.ts:13:20)
}
```

File: universal/hello/contracts/Hello.sol [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import {RevertContext, RevertOptions} from "@zetachain/protocol-contracts/contracts/Revert.sol";
+import "@zetachain/protocol-contracts/contracts/zevm/interfaces/UniversalContract.sol";
+import "@zetachain/protocol-contracts/contracts/zevm/interfaces/IGatewayZEVM.sol";
+import "@zetachain/protocol-contracts/contracts/zevm/interfaces/IZRC20.sol";
+
+contract Hello is UniversalContract {
+    event HelloEvent(string, string);
+    event ContextDataRevert(RevertContext);
+
+    address constant gateway = 0x610178dA211FEF7D417bC0e6FeD39F05609AD788;
```
Comment (andresaiello) [inline universal/hello/contracts/Hello.sol:15]: nit: let's call it gatewayAddress

File: omnichain/swap/tasks/deploy.ts [hunk 2]
```diff
@@ -5,11 +5,11 @@ import { HardhatRuntimeEnvironment } from "hardhat/types";
 const main = async (args: any, hre: HardhatRuntimeEnvironment) => {
   const network = hre.network.name as ParamChainName;
 
-  if (!/zeta_(testnet|mainnet)/.test(network)) {
-    throw new Error(
-      '🚨 Please use either "zeta_testnet" or "zeta_mainnet" network to deploy to ZetaChain.'
-    );
-  }
+  // if (!/zeta_(testnet|mainnet)/.test(network)) {
```
Comment (andresaiello) [inline omnichain/swap/tasks/deploy.ts:8]: if it's no longer valid just delete it, let's try to keep code clean

File: universal/hello/contracts/Hello.sol (hunk 1 above)
Comment (fadeev) [reply universal/hello/contracts/Hello.sol]: f1d8fb4

File: universal/hello/tasks/deploy.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
+import { task } from "hardhat/config";
+import { HardhatRuntimeEnvironment } from "hardhat/types";
+
+const main = async (args: any, hre: HardhatRuntimeEnvironment) => {
+  const network = hre.network.name as ParamChainName;
+
+  // if (!/zeta_(testnet|mainnet)/.test(network)) {
```
[the remark this answers is not in the document]
Comment (fadeev) [reply universal/hello/tasks/deploy.ts:7]: d453ef8

File: omnichain/swap/tasks/deploy.ts (hunk 2 above)
Comment (fadeev) [reply omnichain/swap/tasks/deploy.ts]: Reverted, because it's still relevant: 4095527

**Comment by fadeev** [comment]
@andresaiello please, review again.

**Review by andresaiello — APPROVED** [review]

**Pull request merged**